core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -578,13 +625,24 @@ impl Filters {
 
     /// Retrieves filter from the state.
     pub fn get(&self, index: U256) -> Option<&TypedFilter> {
-        self.state.get(&index)
+        let installed_filter = self.state.get(&index)?;
+
+        Some(&installed_filter.filter)
+    }
+
+    pub fn update_stats(&mut self, index: U256) -> bool {
```

Comment:
I was thinking that it may be possible to pack updating stats and the filter together, but it's actually not the case (and even if it was possible, it probably makes sense to measure intervals between at the start of `getFilterChanges`, not at the end). So combining this logic with `get()` may make sense. A read lock on `Filters` can be acquire faster theoretically, but since in practice we never hold locks, the effect on performance should be minimal.

Reply:
Regarding that, do we need to have  `RwLock`  there at all in case we combine them?

Reply:
I'm thinking maybe we can just use `Mutex`, since we are not doing just read anywhere